### DISCUSSION TITLE
Error while launching code-server

### DISCUSSION THREAD

**Discussion by mistrysiddh** [coder/code-server#3952] [Q&A] [answered]
## OS/Web Information

- Local OS: `Ubuntu 18.04.2 LTS (Bionic Beaver)`
- Remote OS: `Ubuntu 18.04.2 LTS (Bionic Beaver)`
- Remote Architecture: `aarch64`
- `code-server --version`: getting error while fetching
## Log
```
internal/modules/cjs/loader.js:888                      throw err;
  ^
                                                      Error: Cannot find module '/usr/lib/code-server'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:885:15)
    at Function.Module._load (internal/modules/cjs/loader.js:730:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',                             requireStack: []
}
```
this is the error I got when I ran it on `Ubuntu 18.04.2 LTS (Bionic Beaver)`
## Screenshot
![img](https://i.imgur.com/Hd8mbN7.png)

## Notes

This issue can be reproduced in VS Code: Yes

**Comment by balajib-b** [comment]
@mrfoxie if you are submitting a new issue, please fill in the details asked in issue template for better reader understanding

**Comment by mistrysiddh** [comment]
> @mrfoxie if you are submitting a new issue, please fill in the details asked in issue template for better reader understanding

I have changed it

**Comment by jsjoeio** [comment]
This seems like an error related to installation. How did you install code-server? Have you tried installing again?

**Reply by mistrysiddh** [reply]
yes I tried uninstalling and reinstalling code-server but still I get this error. it was working first but now it's not working I tried multiple ways to solve it but the same

**Reply by jsjoeio** [reply]
Hmmm...Can you try uninstalling and reinstalling and letting us know:
- how you're installing it
- an output of the installation logs

Guessing there may be something helpful there

**Reply by mistrysiddh** [reply]
here is the log of installation

```
mistrysiddh@localhost:~$ curl -fsSL https://code-server.dev/install.sh | sh

Ubuntu 18.04.2 LTS
Installing v3.11.1 of the arm64 deb package from GitHub.

+ mkdir -p ~/.cache/code-server
+ curl -#fL -o ~/.cache/code-server/code-server_3.11.1_arm64.deb.incomplete -C - https://github.com/cdr/code-server/releases/download/v3.11.1/code-server_3.11.1_arm64.deb
######################################################################## 100.0%#=#=-#  #                              ######################################################################## 100.0%
+ mv ~/.cache/code-server/code-server_3.11.1_arm64.deb.incomplete ~/.cache/code-server/code-server_3.11.1_arm64.deb
+ sudo dpkg -i ~/.cache/code-server/code-server_3.11.1_arm64.deb
(Reading database ... 17972 files and directories currently installed.)
Preparing to unpack .../code-server_3.11.1_arm64.deb ...
Unpacking code-server (3.11.1) over (3.11.1) ...
Setting up code-server (3.11.1) ...

deb package has been installed.

To have systemd start code-server now and restart on boot:
  sudo systemctl enable --now code-server@$USER
Or, if you don't want/need a background service you can run:
  code-server
mistrysiddh@localhost:~$
```

**Comment by mistrysiddh** [comment]
is there any solution for it?

**Reply by katsie03** [reply]
Is there anything else on the VPS? If not try resetting it, it does wonders. Also FYI if your hoster offers it, upgrade to 20.04. It works first time, also did you manually install or used the script within the repo?

**Reply by mistrysiddh** [reply]
ok thank you but I rather prefer to stay on 18.04.2

**Reply by lg-nineoneninethreenine** [reply]
@mrfoxie try upgrading to Focal Fossa and try again because I use Focal Fossa and it works. It might be problems on Bionic Beaver, maybe thats why.

**Reply by lg-nineoneninethreenine** [reply]
And use the install script.

**Comment by mistrysiddh** [comment]
thank you it's working now

**Reply by jsjoeio** [reply]
Hooray! Which solution ended up working? Can you mark it as the answer?

**Reply by lg-nineoneninethreenine** [reply]
Well, I responded somewhat yesterday as a comment, not as an answer, and I told him to update his Bionic Beaver server to the latest LTS, Focal Fossa. I use it on WSL and it works there. So I told him to upgrade his server to Focal Fossa.

**Comment by mistrysiddh** [comment, marked as answer]
> Is there anything else on the VPS? If not try resetting it, it does wonders. Also FYI if your hoster offers it, upgrade to 20.04. It works first time, also did you manually install or used the script within the repo?

I upgraded my Ubuntu from 18.04.2 to 20.04
